pkg/gardenlet/controller/networkpolicy/helper/helper_test.go

```diff
@@ -113,6 +113,83 @@ var _ = Describe("helper", func() {
 								CIDR: fmt.Sprintf("%s/32", ip4),
 							},
 						},
+						{
+							IPBlock: &networkingv1.IPBlock{
+								CIDR: fmt.Sprintf("%s/32", ip2),
+							},
+						},
+					},
+				},
+			}
+			Expect(egressRules).To(Equal(expectedRules))
+		})
+
+		It("should return the Egress rule with correct IP Blocks of 2 same EndpointSubset", func() {
+			var (
+				ip1     = "10.250.119.142"
+				tcp     = corev1.ProtocolTCP
+				subsets = []corev1.EndpointSubset{
+					{
+						Addresses: []corev1.EndpointAddress{
+							{
+								IP: ip1,
+							},
+						},
+						Ports: []corev1.EndpointPort{
+							{
+								Port:     443,
+								Protocol: tcp,
+							},
+						},
+					},
+					{
+						Addresses: []corev1.EndpointAddress{
+							{
+								IP: ip1,
+							},
+						},
+						Ports: []corev1.EndpointPort{
+							{
+								Port:     443,
+								Protocol: tcp,
+							},
+						},
+					},
+				}
+			)
+
+			egressRules := GetEgressRules(subsets...)
+
+			port443 := intstr.FromInt(443)
+			expectedRules := []networkingv1.NetworkPolicyEgressRule{
+				{
+					To: []networkingv1.NetworkPolicyPeer{
+						{
+							IPBlock: &networkingv1.IPBlock{
+								CIDR: fmt.Sprintf("%s/32", ip1),
+							},
+						},
+					},
+					Ports: []networkingv1.NetworkPolicyPort{
+						{
+							Protocol: &tcp,
+							Port:     &port443,
+						},
+					},
+				},
+				{
+					To: []networkingv1.NetworkPolicyPeer{
+						{
+							IPBlock: &networkingv1.IPBlock{
+								CIDR: fmt.Sprintf("%s/32", ip1),
+							},
+						},
+					},
+					Ports: []networkingv1.NetworkPolicyPort{
+						{
+							Protocol: &tcp,
+							Port:     &port443,
+						},
```

Comment:
How much "more complicated" would the implementation be if we only wanted to have one `NetworkPolicyEgressRule` here instead of having it twice with the same `IPBlock`?

Reply:
It is also possible to check if the `NetworkPolicyEgressRule` has no `IPBlocks` and keep the `existingIPs` outside of the for-loop. 
But I think this is confusing, because I would expect that also the following example will create the expected NetworkPolicy:

```
subset1:
  Addresses:
  - 123.123.123.123
  - 123.123.123.124
  Port:
  - 443
subset2:
  Addresses:
  - 123.123.123.123
  Port:
  - 80
```

With the check for empty IPs this will skip the Rule for `123.123.123.123:80`. I am not sure if this is a use-case for this function, in my opinion the function name suggests that is should work.

WDYT?

Reply:
OK, fair enough, then let's keep it for now as you suggested.